corehq/apps/importer/views.py

```diff
@@ -298,19 +297,17 @@ def importer_job_poll(request, domain, download_id, template="importer/partials/
     try:
         download_context = get_download_context(download_id, check_state=True)
     except TaskFailedError as e:
-        # todo: this is async, so it's totally inappropriate to be using messages.error
-        # todo: and HttpResponseRedirect
         error = e.errors
         if error == 'EXPIRED':
-            return _spreadsheet_expired(request, domain)
+            error = _('Sorry, your session has expired. Please start over and try again.')
         elif error == 'HAS_ERRORS':
-            messages.error(request, _('The session containing the file you '
-                                      'uploaded has expired - please upload '
-                                      'a new one.'))
+            error = _('The session containing the file you uploaded has expired - please upload a new one.')
```

Comment:
It seems like this error message might not be accurate (although I appreciate you're just keeping the original error message). `HAS_ERRORS` occurs on `ImporterError`, which could be raised on file not found, or if there is a problem with the spreadsheet itself. If we are checking explicitly for `ImporterRefError`, we might want to check for `ImporterFileNotFound` and `ImporterExcelError` separately too. Just a suggestion though. Not blocking.